QStore4S/src/main/java/edu/asu/qstore4s/service/impl/RepositoryManager.java

```diff
@@ -178,6 +185,29 @@ public String getList(String xml, String accept) throws ParserException, IOExcep
     /**
      * {@inheritDoc}
      * 
+     * @throws ExecutionException
+     * 
+     * @throws JSONException
+     */
+    @Override
+    @Async
+    public void executeQueryAsync(String query, Class<?> clazz, Integer queryID) throws ExecutionException {
+
+        asyncQueryManager.setQueryStatus(queryID, ExecutionStatus.RUNNING);
+
+        try {
+            List<CreationEvent> elementList = storeManager.executeQuery(query, clazz);
+
+            asyncQueryManager.setQueryResult(queryID, elementList);
+            asyncQueryManager.setQueryStatus(queryID, ExecutionStatus.COMPLETED);
+        } catch (Exception e) {
```

Comment:
The methods throw no execeptions, I just wanted to change the status of  the query in case there is a RuntimeException.